src/test/index.ts

```diff
@@ -31,8 +31,86 @@ type TestFixtures = PlaywrightTestArgs & PlaywrightTestOptions & {
 type WorkerAndFileFixtures = PlaywrightWorkerArgs & PlaywrightWorkerOptions & {
   _browserType: BrowserType;
   _artifactsDir: () => string,
+  _reuseBrowerContext: ReuseBrowerContextStorage,
 };
 
+class ReuseBrowerContextStorage {
+  private _browserContext?: BrowserContext;
+  private _uniqueOrigins = new Set<string>();
+
+  isEnabled(): boolean {
+    return !!process.env.PWTEST_REUSE_CONTEXT;
+  }
+
+  hasContext(): boolean {
+    return !!this._browserContext;
+  }
+
+  setContext(context: BrowserContext) {
+    assert(!this._browserContext);
+    this._browserContext = context;
+    this._browserContext.on('page', page => page.on('framenavigated', frame => this._uniqueOrigins.add(new URL(frame.url()).origin)));
+  }
+
+  async obtainContext(newContextOptions: BrowserContextOptions): Promise<BrowserContext> {
+    assert(this._browserContext);
+    let pages = this._browserContext.pages();
+    const page = pages[0];
+    if (this._uniqueOrigins.size > 0) {
+      let pageWorker = null;
+      if (pages.length > 1) {
+        await page.bringToFront();
+        pageWorker = pages[1];
+      } else {
+        pageWorker = await this._browserContext.newPage();
+        pages = this._browserContext.pages();
+      }
+      for (const origin of this._uniqueOrigins) {
+        await pageWorker.route(origin, route => route.fulfill({body: `<html></html>`, contentType: 'text/html'}), { times: 1 });
+        await pageWorker.goto(origin);
+        await pageWorker.evaluate(() => window.localStorage.clear());
+        await pageWorker.evaluate(() => window.sessionStorage.clear());
+      }
+    }
+    for (const p of pages.slice(1))
```

Comment:
You should have done it earlier - all these pages are polluting localStorage while they are alive. You clear it above, but it is not clean again.

Reply:
done.